sklearn/tests/test_calibration.py

```diff
@@ -341,3 +348,31 @@ def decision_function(self, X):
     calibrated_clf = CalibratedClassifierCV(MockTensorClassifier())
     # we should be able to fit this classifier with no error
     calibrated_clf.fit(X, y)
+
+
+def test_calibration_pipeline():
+    # Test that calibration works in pre-fit pipeline with transformer,
+    # where X is not array-like, sparse matrix or dataframe at the start. See
+    # issue #8710
+    fake_features = [
+        {'state': 'NY', 'age': 'adult'},
+        {'state': 'TX', 'age': 'adult'},
+        {'state': 'VT', 'age': 'child'},
+    ]
+    labels = [1, 0, 1]
+
+    pipeline = Pipeline([
+                ('vectorizer', DictVectorizer()),
+                ('clf', RandomForestClassifier())
+        ])
+    pipeline.fit(fake_features, labels)
+
+    calib_clf = CalibratedClassifierCV(pipeline, cv='prefit')
```

Comment:
So if we want to use the `cv=None` with the same pipeline, it is not going to work anyway?